scenario-api/src/test/scala/pl/touk/nussknacker/engine/graph/NodeDataGen.scala

```diff
@@ -0,0 +1,233 @@
+package pl.touk.nussknacker.engine.graph
+
+import org.scalacheck.{Arbitrary, Gen}
+import pl.touk.nussknacker.engine.api.LayoutData
+import pl.touk.nussknacker.engine.api.parameter.ParameterName
+import pl.touk.nussknacker.engine.graph.evaluatedparam.{BranchParameters, Parameter}
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.engine.graph.expression.Expression.Language
+import pl.touk.nussknacker.engine.graph.fragment.FragmentRef
+import pl.touk.nussknacker.engine.graph.node.FragmentInputDefinition.{FragmentClazzRef, FragmentParameter}
+import pl.touk.nussknacker.engine.graph.node.{
+  BranchEndData,
+  BranchEndDefinition,
+  CustomNode,
+  Enricher,
+  Filter,
+  FragmentInput,
+  FragmentInputDefinition,
+  FragmentOutputDefinition,
+  Join,
+  NodeData,
+  Processor,
+  Source,
+  Split,
+  Switch,
+  UserDefinedAdditionalNodeFields,
+  Variable,
+  VariableBuilder
+}
+import pl.touk.nussknacker.engine.graph.service.ServiceRef
+import pl.touk.nussknacker.engine.graph.source.SourceRef
+import pl.touk.nussknacker.engine.graph.variable.Field
+
+class NodeDataGen private {
```

Comment:
WDYT about moving it to `pl.touk.nussknacker.test.utils.generators` package?

Reply:
But then it will introduce a dependency of `test-utils` on `scenario-api` to get to `NodeData` implementations, no? And currently `scenario-api` is dependent on `test-utils` during test. Or do I understand it in a wrong way?